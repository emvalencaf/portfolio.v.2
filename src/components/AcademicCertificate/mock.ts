import { AcademicCertificateProps } from ".";

export default {
	courseTitle: "Curso de React.Js e Next.Js (Intermediário e Avançado)",
	institution: "Udemy, Instrutor: Luiz Otávio Miranda",
	courseWork: "93 horas",
	startIn: 2932983,
	endIn: 3932983,
	urlDownload: "/assets/files/UC-44225967-b7ec-4add-8983-426364ad7bb6.pdf"
} as AcademicCertificateProps;