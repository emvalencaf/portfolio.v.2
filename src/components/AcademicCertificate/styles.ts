import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;