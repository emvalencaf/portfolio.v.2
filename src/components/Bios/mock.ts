import { BiosProps } from ".";

export default {
	bios: `
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore rem minus dolore ducimus quos, sed dolores odit maiores quam tempora atque deserunt voluptas perferendis, aspernatur, sequi dignissimos reiciendis. In, esse!
	Lorem ipsum, dolor sit amet consectetur adipisicing elit. Non a deleniti quia. Magnam aspernatur cumque, nemo vero libero excepturi consequuntur corporis ullam nobis molestiae? Atque nam praesentium ullam quaerat quisquam?
</p>
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio cumque recusandae libero sequi, tempora iste, illo, possimus unde cum numquam quae corrupti minima tenetur mollitia earum repellendus animi molestiae voluptates.
	Lorem ipsum dolor sit, amet consectetur adipisicing elit. Obcaecati dolores fugiat ipsam ut odit sed aliquid ducimus molestiae vero ipsa nam veniam dolor libero labore qui pariatur, alias assumenda itaque?
	Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo cum temporibus nesciunt sequi odit ipsam harum magnam dolore a molestiae. Totam debitis, rerum aliquid sit non ratione. Et, omnis iusto!
</p>
<p>
	Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nam dicta aut cum sequi voluptatibus. Vel pariatur veritatis quidem. Culpa vero deleniti minima earum quod, odit deserunt fugit reprehenderit architecto in!
</p>
	`,
	profilePhoto: {
		altText: "profile picture",
		srcImg: "assets/images/potrait-photo.jpg",
	},
} as BiosProps;
