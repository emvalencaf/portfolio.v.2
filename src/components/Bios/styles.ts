import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
