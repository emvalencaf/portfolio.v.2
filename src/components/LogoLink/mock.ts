import { LogoLinkProps } from ".";

export default {
	srcImg: "assets/images/logo.png",
	link: "#",
	altText: "logo"
} as LogoLinkProps;
