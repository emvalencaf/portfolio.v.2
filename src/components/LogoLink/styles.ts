import styled, { css } from 'styled-components';

export const Link = styled.a`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		text-decoration: none;
		color: inherit;
		> img {
			width: ${theme.fonts.sizes.xhuge};
			height: ${theme.fonts.sizes.xhuge};
			cursor: pointer;
		}
	`}
`;
