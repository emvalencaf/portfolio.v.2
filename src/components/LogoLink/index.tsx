// components
import Link from 'next/link';

// styles
import * as Styled from './styles';

// types
export type LogoLinkProps = {
	link: string;
	altText: string;
	srcImg?: string;
	newTab?: boolean;
};

const LogoLink = ({ link, srcImg = "", altText, newTab = false }: LogoLinkProps) => {

	const target = newTab ? "_blank" : "_self";
	const hostname = location.hostname;
	const rel = link.includes(hostname) || link.match(/^[\/\#]/) ? "internal" : "external";
	const nextLink = link.match(/^\//) ? true : false;

	if (nextLink) return (
		<Link href={link} legacyBehavior passHref>
			<Styled.Link target={target} rel={rel} aria-label="logo link">
					{srcImg ? <img src={srcImg} alt={altText} /> : altText}
			</Styled.Link>
		</Link>
	);

	return (
		<Styled.Link target={target} rel={rel} aria-label="logo link">
			{srcImg ? <img src={srcImg} alt={altText} /> : altText}
		</Styled.Link>
	);
};

export default LogoLink;
