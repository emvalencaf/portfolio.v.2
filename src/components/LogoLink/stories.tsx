import { Meta, Story } from "@storybook/react/types-6-0";
import LogoLink, { LogoLinkProps } from ".";

// mock
import mock from "./mock";

export default {
	title: "LogoLink",
	component: LogoLink,
	args: mock,
	argTypes: {
		srcImg: { type: "string" },
		link: { type: "string" },
		altText: { type: "string" },
		newTab: { type: "boolean" },
	},
} as Meta<LogoLinkProps>;

export const Template: Story<LogoLinkProps> = (args) => {
	return (
		<div>
			<LogoLink {...args} />
		</div>
	);
};
