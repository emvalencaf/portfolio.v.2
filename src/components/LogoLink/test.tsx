import { logDOM, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

// react component to be test
import LogoLink from '.';

// mock
import mock from "./mock";

describe('<LogoLink />', () => {
	it('should render LogoLink with an image', () => {
		renderTheme(<LogoLink {...mock} />);
		const img = screen.getByRole("img", { name: mock.altText});
		const link = img.parentElement;
		expect(img).toBeInTheDocument();
		expect(link).toHaveAttribute("target", "_self");
	});

	it('should render LogoLink with only text', () => {
		const { container } = renderTheme(<LogoLink link="https://wwww.google.com.br/" altText={mock.altText}/>);
		const link = screen.getByText(mock.altText);

		expect(screen.queryByRole("img", { name: mock.altText})).not.toBeInTheDocument();
		expect(link).toHaveAttribute("rel","external");
	});


	it('should render a LogoLink with a next link', () => {
		const { container } = renderTheme(<LogoLink link="/" altText={mock.altText}/>);
		const link = screen.getByText(mock.altText);

		expect(screen.queryByRole("img", { name: mock.altText})).not.toBeInTheDocument();
		expect(link).toHaveAttribute("rel","internal");
	});

	it("should render LogoLink with link that will open a new tab", () => {
		renderTheme(<LogoLink link='/home' altText={mock.altText} srcImg={mock.srcImg} newTab />);

		expect(screen.queryByRole("link")).toHaveAttribute("target","_blank");
		expect(screen.getByRole("img", { name: mock.altText})).toBeInTheDocument();
		expect(screen.queryByText(mock.altText)).not.toBeInTheDocument();

	})

	it('should match snapshot', () => {
		const { container } = renderTheme(<LogoLink {...mock} />);
		expect(container).toMatchSnapshot();
	});
});
