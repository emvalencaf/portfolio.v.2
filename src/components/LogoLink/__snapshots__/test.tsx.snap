// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c0 > img {
  width: 6.4rem;
  height: 6.4rem;
  cursor: pointer;
}

<div>
  <a
    aria-label="logo link"
    class="c0"
    rel="internal"
    target="_self"
  >
    <img
      alt="logo"
      src="assets/images/logo.png"
    />
  </a>
</div>
`;
